asf_welsh_energy_consultation/analysis/produce_plots_and_stats.py

```diff
@@ -91,7 +91,7 @@
                 # domain ensures good margin at left/right of chart
                 "year",
                 title="Year",
-                scale=alt.Scale(domain=["2007-07-01", "2023-01-01"]),
+                scale=alt.Scale(domain=["2007-07-01", f"{max_date.year}-07-01"]),
```

Comment:
yes this will always be the same because this graph only takes data for each full year and not partial years. This domain just helps to create a good margin on the right side of chart (see below)
![image](https://github.com/nestauk/asf_welsh_energy_consultation/assets/104171770/e49523c1-bee5-40b6-9175-72b312fcb535)